Amazon.QLDB.Driver/session/AsyncQldbSession.cs

```diff
@@ -14,19 +14,187 @@
 namespace Amazon.QLDB.Driver
 {
     using System;
+    using System.Net;
+    using System.Threading;
     using System.Threading.Tasks;
+    using Amazon.QLDBSession.Model;
+    using Amazon.Runtime;
+    using Microsoft.Extensions.Logging;
 
-    internal class AsyncQldbSession
+    /// <summary>
+    /// <para>Represents a session to a specific ledger within QLDB, allowing for asynchronous execution of PartiQL statements and
+    /// retrieval of the associated results, along with control over transactions for bundling multiple executions.</para>
+    ///
+    /// <para>The execute methods provided will automatically retry themselves in the case that an unexpected recoverable error
+    /// occurs, including OCC conflicts, by starting a brand new transaction and re-executing the statement within the new
+    /// transaction.</para>
+    ///
+    /// </summary>
+    internal class AsyncQldbSession : BaseQldbSession
     {
-        public void Release()
+        private readonly ILogger logger;
+        private readonly Action<AsyncQldbSession> releaseSession;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AsyncQldbSession"/> class.
+        /// </summary>
+        ///
+        /// <param name="session">The session object representing a communication channel with QLDB.</param>
+        /// <param name="releaseSession">The delegate method to release the session.</param>
+        /// <param name="logger">The logger to be used by this.</param>
+        internal AsyncQldbSession(Session session, Action<AsyncQldbSession> releaseSession, ILogger logger)
         {
-            throw new NotImplementedException();
+            this.session = session;
+            this.releaseSession = releaseSession;
+            this.logger = logger;
+            this.isAlive = true;
```

Comment:
this.isAlive and this.session, as well as this.logger seems like logic that can live in the base session constructor

Reply:
Done.